packages/trade/exchange/migrations/1625819983722-DirectionAndStatusStrings.ts

```diff
@@ -0,0 +1,100 @@
+import { MigrationInterface, QueryRunner } from 'typeorm';
+
+import { TransferStatus } from '../src/pods/transfer/transfer-status';
+import { TransferDirection } from '../src/pods/transfer/transfer-direction';
+
+const getNewTransferDirection = (oldDirection: number): TransferDirection => {
+    switch (oldDirection) {
+        case 0:
+            return TransferDirection.Deposit;
+        case 1:
+            return TransferDirection.Withdrawal;
+        case 2:
+            return TransferDirection.Claim;
+        case 3:
+            return TransferDirection.Send;
+    }
+};
+
+const getNewTransferStatus = (oldStatus: number): TransferStatus => {
+    switch (oldStatus) {
+        case 0:
+            return TransferStatus.Unknown;
+        case 1:
+            return TransferStatus.Accepted;
+        case 2:
+            return TransferStatus.Unconfirmed;
+        case 3:
+            return TransferStatus.Confirmed;
+        case 4:
+            return TransferStatus.Error;
+        default:
+            return TransferStatus.Unknown;
+    }
+};
+
+export class DirectionAndStatusStrings1625819983722 implements MigrationInterface {
+    name = 'DirectionAndStatusStrings1625819983722';
+
+    public async up(queryRunner: QueryRunner): Promise<void> {
+        const oldTransfers = await queryRunner.query(
+            `SELECT id, status, direction FROM "exchange_transfer"`
+        );
+
+        await queryRunner.query(`ALTER TABLE "exchange_transfer" DROP COLUMN "status"`);
+        await queryRunner.query(`ALTER TABLE "exchange_transfer" DROP COLUMN "direction"`);
+        await queryRunner.query(`ALTER TABLE "exchange_transfer" ADD "status" character varying`);
+        await queryRunner.query(
+            `ALTER TABLE "exchange_transfer" ADD "direction" character varying`
```

Comment:
so you drop this column and add again? To clean up?

Reply:
Yeah. I thought about using `ALTER TABLE ... TYPE ... USING <expression>` but I couldn't find the correct expression to use there.

Since the migration is a transaction that is commited only after the `up()` method is fully executed, I think this approach should work fine.